contracts/helper/RebalancingSetCTokenBidder.sol

```diff
@@ -0,0 +1,357 @@
+        depositComponents(
+            combinedTokenArray,
+            inflowArray
+        );
+
+        // Bid in auction
+        rebalanceAuctionModule.bidAndWithdraw(
+            address(_rebalancingSetToken),
+            _quantity,
+            _allowPartialFill
+        );
+
+        // Withdraw auction tokens or underlying to user
+        withdrawComponentsToSender(
+            combinedTokenArray
+        );
+
+        // Log bid placed with Eth event
+        emit BidPlacedCToken(
+            address(_rebalancingSetToken),
+            msg.sender
+        );
+    }
+
+    /*
+     * Get token inflows and outflows and combined token array denominated in underlying required
+     * for bid for a given rebalancing Set token.
+     *
+     * @param _quantity               The amount of currentSet to be rebalanced
+     * @param _rebalancingSetToken    The rebalancing Set Token instance
+     * @return inflowArray            Array of amount of tokens inserted into system in bid
+     * @return outflowArray           Array of amount of tokens returned from system in bid
+     * @return combinedTokenArray     Array of token addresses
+     */
+    function getBidPriceAndAddressArray(
+        uint256 _quantity,
+        IRebalancingSetToken _rebalancingSetToken
+    )
+        public
+        view
+        returns (uint256[] memory, uint256[] memory, address[] memory)
+    {
+        // Get token addresses
+        address[] memory combinedTokenArray = _rebalancingSetToken.getCombinedTokenArray();
+
+        // Get inflow and outflow arrays for the given bid quantity
+        uint256[] memory inflowArray;
+        uint256[] memory outflowArray;
+        (
+            inflowArray,
+            outflowArray
+        ) = _rebalancingSetToken.getBidPrice(_quantity);
+
+        // Loop through the combined token addresses array and replace with underlying address
+        for (uint256 i = 0; i < combinedTokenArray.length; i++) {
+            address currentComponent = combinedTokenArray[i];
+
+            if (cTokenToUnderlying[currentComponent] != address(0)) {
+                combinedTokenArray[i] = cTokenToUnderlying[currentComponent];
+
+                // Replace inflow and outflow with required amount of underlying. 
+                // Calculated as cToken quantity * exchangeRate / 10 ** 18.
+                uint256 exchangeRate = ICToken(currentComponent).exchangeRateStored();
+                uint256 currentInflowQuantity = inflowArray[i];
+                uint256 currentOutflowQuantity = outflowArray[i];
+
+                inflowArray[i] = currentInflowQuantity.mul(exchangeRate).div(CommonMath.scaleFactor());
+                outflowArray[i] = currentOutflowQuantity.mul(exchangeRate).div(CommonMath.scaleFactor());
+
+                // Check for rounding error and add 1 if needed
+                inflowArray[i] = inflowArray[i].mul(CommonMath.scaleFactor()).div(exchangeRate) >= currentInflowQuantity
```

Comment:
To simplify this, use `CommonMath. divCeil` instead - which I think achieves what you want? You can even do it in the above function.

Reply:
+1